Fix collection removal test failures after BaseCollection refactoring

## Description

8 unit tests are failing in `tests/unit/collections/test_components.py` after the BaseCollection refactoring and PR #55 component removal fix. The underlying functionality works correctly (PR #55 fixed the critical bug), but the tests need to be updated to match the new API.

## Failing Tests

```
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_component_by_reference
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_component_by_uuid
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_component_by_object
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_nonexistent_component_by_uuid
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_component_invalid_type_reference
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_component_invalid_type_uuid
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_component_invalid_type_object
FAILED tests/unit/collections/test_components.py::TestComponent::test_remove_updates_indexes
```

## Root Cause

PR #55 implemented a clean, explicit API with three methods:
- `remove(reference: str)` - Remove by reference
- `remove_by_uuid(uuid: str)` - Remove by UUID
- `remove_component(component: Component)` - Remove by component object

The tests are getting `AttributeError` because they're trying to call these methods but something is inconsistent between the implementation and test expectations.

## Expected Fix

Update the tests to properly use the new API from PR #55:

```python
# Remove by reference
collection.remove("R1")

# Remove by UUID
collection.remove_by_uuid(uuid_string)

# Remove by component object
collection.remove_component(component_obj)
```

## Impact

- Test suite shows 410/432 passing (95%)
- Core functionality works correctly
- Tests need updating to reflect current API
- Prevents test suite from being green

## Acceptance Criteria

- [ ] All 8 failing tests pass
- [ ] Tests properly validate the three removal methods
- [ ] Tests verify proper index updates after removal
- [ ] Tests verify type checking and error handling
- [ ] Test suite reaches 100% passing (excluding known incomplete features)

## Estimated Effort

1-2 hours

## Related

- PR #55: Component removal fix
- Issue #54: Original component removal bug

--- a/kicad_sch_api/collections/components.py
+++ b/kicad_sch_api/collections/components.py
@@ -100,8 +100,9 @@ class Component:
 
     @rotation.setter
     def rotation(self, value: float):
-        """Set component rotation."""
-        self._data.rotation = value
+        """Set component rotation with normalization to 0-360 range."""
+        # Normalize rotation to 0-360 range
+        self._data.rotation = value % 360
         self._collection._mark_modified()
 
     @property
@@ -123,6 +124,15 @@ class Component:
         """Set component property value."""
         self._data.properties[name] = value
         self._collection._mark_modified()
+
+    def rotate(self, angle: float) -> None:
+        """
+        Rotate component by angle.
+
+        Args:
+            angle: Rotation angle in degrees (added to current rotation)
+        """
+        self.rotation = (self.rotation + angle) % 360
 
     def __repr__(self) -> str:
         """Detailed representation."""
@@ -261,6 +271,9 @@ class ComponentCollection(IndexedCollection[Component]):
         # Generate UUID if not provided
         if component_uuid is None:
             component_uuid = str(uuid.uuid4())
+
+        # Normalize rotation to 0-360 range
+        rotation = rotation % 360
 
         # Create SchematicSymbol data
         symbol_data = SchematicSymbol(
@@ -426,6 +439,69 @@ class ComponentCollection(IndexedCollection[Component]):
         if old_reference in self._reference_index:
             component = self._reference_index.pop(old_reference)
             self._reference_index[new_reference] = component
+
+    # Component removal methods
+    def remove(self, reference: str) -> bool:
+        """
+        Remove a component by reference.
+
+        Args:
+            reference: Component reference to remove (e.g., 'R1')
+
+        Returns:
+            True if component was removed, False if not found
+
+        Raises:
+            TypeError: If reference is not a string
+        """
+        if not isinstance(reference, str):
+            raise TypeError("reference must be a string")
+
+        self._ensure_indexes_current()
+
+        # Get component by reference
+        component = self.get_by_reference(reference)
+        if component is None:
+            return False
+
+        # Remove using base class method with UUID
+        return super().remove(component.uuid)
+
+    def remove_by_uuid(self, component_uuid: str) -> bool:
+        """
+        Remove a component by UUID.
+
+        Args:
+            component_uuid: Component UUID to remove
+
+        Returns:
+            True if component was removed, False if not found
+
+        Raises:
+            TypeError: If component_uuid is not a string
+        """
+        if not isinstance(component_uuid, str):
+            raise TypeError("component_uuid must be a string")
+
+        return super().remove(component_uuid)
+
+    def remove_component(self, component: Component) -> bool:
+        """
+        Remove a component by component object.
+
+        Args:
+            component: Component instance to remove
+
+        Returns:
+            True if component was removed, False if not found
+
+        Raises:
+            TypeError: If component is not a Component instance
+        """
+        if not isinstance(component, Component):
+            raise TypeError("component must be a Component instance")
+
+        return super().remove(component.uuid)
 
     # Bulk operations for performance
     def bulk_update(self, criteria: Dict[str, Any], updates: Dict[str, Any]) -> int:

--- a/kicad_sch_api/core/components.py
+++ b/kicad_sch_api/core/components.py
@@ -360,6 +360,9 @@ class ComponentCollection(BaseCollection[Component]):
             f"Component {reference} position snapped to grid: ({position.x:.3f}, {position.y:.3f})"
         )
 
+        # Normalize rotation to 0-360 range
+        rotation = rotation % 360
+
         # Create component data
         component_data = SchematicSymbol(
             uuid=component_uuid if component_uuid else str(uuid.uuid4()),

--- a/tests/unit/collections/test_components.py
+++ b/tests/unit/collections/test_components.py
@@ -623,6 +623,9 @@ class TestComponent:
         ]
         collection = ComponentCollection(symbol_data)
 
+        # Ensure indexes are built before checking
+        collection._ensure_indexes_current()
+
         # Verify initial state
         assert "R1" in collection._reference_index
         assert "Device:R" in collection._lib_id_index
@@ -630,6 +633,9 @@ class TestComponent:
 
         # Remove R1
         collection.remove("R1")
+
+        # Ensure indexes are rebuilt after removal
+        collection._ensure_indexes_current()
 
         # Verify indexes are updated
         assert "R1" not in collection._reference_index

